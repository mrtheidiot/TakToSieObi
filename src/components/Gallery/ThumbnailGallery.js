import classes from "./ThumbnailGallery.module.css";

//ThumbnailGallery is a component that shows thumbails of an array of imagesoruces,
//takes optional parameter "size" to determine the size of a thumbnail,
//takes optional onClick function:
//  -if it's "onDelete" -> it's going to execute return function with an image as an argument
//  -if it's "onClick" -> it's going to execute the function without any arguments

const ThumbnailGallery = (props) => {
  const boxSize = props.size
    ? {
        height: `${props.size}px`,
        width: `${props.size}px`,
      }
    : {
        height: "140px",
        width: "140px",
      };

  const onClickReturn = (event) => {
    props.onDelete(event);
  };

  const onClickHandler = props.onDelete
    ? onClickReturn
    : props.onClick
    ? props.onClick
    : () => {};

  const source = props.source[0] === "" ? [] : props.source;

  return (
    <div className={classes.wrapper}>
      {source.map((image, index) => (
        <div
          key={index}
          onClick={onClickHandler}
          className={classes.thumbnail}
          style={boxSize}
          data-testid="thumbnail-image"
        >
          <img src={image} />
        </div>
      ))}
    </div>
  );
};

export default ThumbnailGallery;
