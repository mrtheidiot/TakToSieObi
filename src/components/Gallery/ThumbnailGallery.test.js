import { render } from "@testing-library/react";
import placeholder from './../../assets/placeholder.png'
import ThumbnailGallery from "./ThumbnailGallery";

it("matches the UploadBar snapshot", () => {
  const props = {
    source: [placeholder],
  };

  const {
    container: { firstChild },
  } = render(<ThumbnailGallery {...props} />);

  expect(firstChild).toMatchSnapshot();
});
