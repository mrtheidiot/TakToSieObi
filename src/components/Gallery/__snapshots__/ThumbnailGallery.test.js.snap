// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the UploadBar snapshot 1`] = `
<div
  class="wrapper"
>
  <div
    class="thumbnail"
    data-testid="thumbnail-image"
    style="height: 140px; width: 140px;"
  >
    <img
      src="placeholder.png"
    />
  </div>
</div>
`;
